docs/docs/adding-pagination.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Adding pagination
+---
+
+A page displaying a list of content gets longer as the amount of content grows.
+Pagination is the technique of spreading that content across multiple pages.
+
+The goal is to create multiple pages (from a single [template](https://www.gatsbyjs.org/docs/building-with-components/#page-template-components)) that show a limited number of items.
+
+Each page will [query GraphQL](https://www.gatsbyjs.org/docs/querying-with-graphql/) for the items it wants to display.
+
+The information needed to query for those specific items will come from the [`context`](https://www.gatsbyjs.org/docs/graphql-reference/#query-variables) that is added when [creating pages](https://www.gatsbyjs.org/docs/creating-and-modifying-pages/#creating-pages-in-gatsby-nodejs) in `gatsby-node`
+
+### Example
+
+```js{22-27}
+// src/templates/blog-list-template.js
```

Comment:
we can use use something like `js{22-27}:title=src/templates/blog-list-template.js` to get nice title bar in codeblock that looks like this:
![screen shot 2018-09-24 at 13 14 09](https://user-images.githubusercontent.com/419821/45949309-cc4b8d00-bffb-11e8-8115-3abbfd877035.png)